### PR TITLE
Implemented ShoppingCartController, UserController, OrderController, CinemaHallController

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

**Review by viaviaj — APPROVED** [review]
Well done!

File: src/main/java/com/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public OrderController(OrderService orderService, ShoppingCartService shoppingCartService,
+                           UserService userService, OrderDtoMapper mapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping
+    public void addOrder(@RequestBody Long userId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/controller/OrderController.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/homework/lesson_33_2_hw.html

File: src/main/java/com/cinema/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    private final UserDtoMapper mapper;
+
+    public UserController(UserService userService, UserDtoMapper mapper) {
+        this.userService = userService;
+        this.mapper = mapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getUserByEmail(@RequestParam String email) {
+        return mapper.getUserResponseDto(userService.findByEmail(email).get());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/controller/UserController.java]:
```suggestion
        return mapper.getUserResponseDto(userService.findByEmail(email).get());
```
userService returning user(not optional). Let make userService.findByEmail() method return user too

File: src/main/java/com/cinema/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -55,4 +55,12 @@ public User add(User user) {
             return query.uniqueResultOptional();
         }
     }
+
+    @Override
+    public User get(Long id) {
+        logger.info("Trying to get User with id: " + id);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/dao/impl/UserDaoImpl.java]: if you are using lombok in your poroject, let's use it for loggin too

File: src/main/java/com/cinema/model/dto/cart/ShoppingCartResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.model.dto.cart;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsIds;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/model/dto/cart/ShoppingCartResponseDto.java]: let's add userId

File: src/main/java/com/cinema/model/dto/order/OrderResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.cinema.model.Ticket;
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderResponseDto {
+    private Long id;
+    private List<Ticket> tickets;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/model/dto/order/OrderResponseDto.java]: why list of Tickets?

File: src/main/java/com/cinema/controller/OrderController.java [hunk 6]
```diff
@@ -32,7 +32,7 @@ public OrderController(OrderService orderService, ShoppingCartService shoppingCa
     }
 
     @PostMapping
-    public void addOrder(@RequestBody Long userId) {
+    public void completeOrder(@RequestBody Long userId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/controller/OrderController.java:35]: add "/complete" to url

File: src/main/java/com/cinema/Main.java [hunk 7]
```diff
@@ -17,12 +17,11 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
-import org.apache.log4j.Logger;
+import lombok.extern.log4j.Log4j;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
+@Log4j
 public class Main {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/Main.java]: make a collection of requests in postman to fill your database and remove the main class

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**